src/core/agent/services/identifierService.ts

```diff
@@ -375,6 +377,86 @@ class IdentifierService extends AgentService {
     });
     return multisigId;
   }
+
+  async rotateMultisig(ourIdentifier: string): Promise<string | undefined> {
+    const metadata = await this.getMetadataById(ourIdentifier);
+    if (metadata.method !== IdentifierType.KERI) {
+      throw new Error(IdentifierService.ONLY_CREATE_ROTATION_WITH_AID);
+    }
+    if (!metadata.signifyName) {
+      throw new Error(IdentifierService.AID_MISSING_SIGNIFY_NAME);
+    }
+    // Make sure no non-Keri contacts get passed into this function
+    const members = await this.agent.modules.signify.getMultisigMembers(
+      metadata.signifyName
+    );
+    const multisigMembers: Aid[] = [];
+    await Promise.all(
+      members?.signing?.map(async (signing: any) => {
+        const aid = await this.agent.modules.signify.queryKeyState(
+          signing.aid,
+          "0"
+        );
+        multisigMembers.push({ state: aid.response } as Aid);
+      })
+    );
+    const identifiers = await this.getIdentifiers();
+    const identifier = identifiers.find((identifier) => {
+      return multisigMembers.find((item) => identifier.id === item.state.i);
+    });
+    if (identifier && identifier.signifyName) {
+      const aid = await this.agent.modules.signify.getIdentifierByName(
+        identifier?.signifyName
+      );
+
+      const result = await this.agent.modules.signify.rotateMultisigAid(
+        aid,
+        multisigMembers,
+        metadata.signifyName
+      );
+      const multisigId = result.op.name.split(".")[1];
+      return multisigId;
+    }
+  }
+
+  async joinMultisigRotation(
+    notification: KeriNotification
+  ): Promise<string | undefined> {
+    const msgSaid = notification.a.d as string;
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+    if (!notifications.length) {
+      throw new Error(IdentifierService.SAID_NOTIFICATIONS_NOT_FOUND);
+    }
+    const exn = notifications[0].exn;
+    const multisigId = exn.a.gid;
+    const multiSig = await this.agent.modules.signify.getIdentifierById(
+      multisigId
+    );
+    if (!multiSig) {
+      throw new Error(IdentifierService.MULTI_SIG_NOT_FOUND);
+    }
+    const rstate = exn.a.rstates;
+    const identifiers = await this.getIdentifiers();
+    const identifier = identifiers.find((identifier) => {
+      return rstate.find((item) => identifier.id === item.i);
+    });
```

Comment:
it mean when we create multi-sig, we must store the AID that control this multi-sig, right? I think we still need to filter it when creating the multi-sig, because the exchange message is not pointed to specified AID

Reply:
Ahh... OK, then can we filter in the SQL? Filter by ID

Reply:
Yep, we had the function to filter with ID